lib/core/mergeConfig.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var utils = require('../utils');
+
+/**
+ * Config-specific merge-function which creates a new config-object
+ * based on given defaults and instance config.
+ *
+ * @param {Object} config1
+ * @param {Object} config2
+ * @returns {Object} New object resulting from merging config2 to config1
+ */
+module.exports = function mergeConfig(config1, config2) {
+  // eslint-disable-next-line no-param-reassign
+  config2 = config2 || {};
+  var config = {};
+
+  utils.forEach(['url', 'method', 'params', 'data'], function valueFromInstanceConfig(prop) {
+    if (typeof config2[prop] !== 'undefined') {
```

Comment:
I've just realized that we always want to use `url`, `method`, `params`, and `data` values from the request config (*not* from `defaults` or the instance config). It means that we should probably remove the `if (typeof config2[prop] !== 'undefined')` check. Sorry for the back and forth!

Reply:
`valueFromInstanceConfig` name is misleading. `valueFromConfig2`?